src/Package/Impl/Debugger/DataTips/DataTipTextViewConnectionListener.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.ObjectModel;
+using System.ComponentModel.Composition;
+using Microsoft.R.Components.ContentTypes;
+using Microsoft.VisualStudio.R.Package.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.R.Package.Debugger.DataTips {
+    [Export(typeof(IWpfTextViewConnectionListener))]
+    [ContentType(RContentTypeDefinition.ContentType)]
+    [TextViewRole(PredefinedTextViewRoles.Debuggable)]
+    internal sealed class DataTipTextViewConnectionListener : IWpfTextViewConnectionListener {
+        public DataTipTextViewConnectionListener() {
+        }
+
+        public void SubjectBuffersConnected(IWpfTextView textView, ConnectionReason reason, Collection<ITextBuffer> subjectBuffers) {
+            var debugger = VsAppShell.Current.GetGlobalService<IVsDebugger>();
+            new DataTipTextViewFilter(textView, debugger);
+        }
+
+        public void SubjectBuffersDisconnected(IWpfTextView textView, ConnectionReason reason, Collection<ITextBuffer> subjectBuffers) {
```

Comment:
The corresponding code in PTVS never removes it (and I guess the check for content type that's done first thing in the filter before anything else makes it work). 

So, how to do this correctly? Check for content type of the buffer, and if it's not R, remove the filter? Also, where would be the appropriate place to store the reference to the filter so it can be passed to `RemoveCommandFilter`? Does `textView` has some kind of property bag, or let you enumerate all filters so that I could just use `is` to check for mine?